x-pack/plugins/cases/public/components/case_view/index.tsx

```diff
@@ -5,480 +5,24 @@
-import { getConnectorById } from '../utils';
+import { CasesTimelineIntegrationProvider } from '../timeline_context';
 import { DoesNotExist } from './does_not_exist';
 import { useKibana } from '../../common/lib/kibana';
 import { useCasesContext } from '../cases_context/use_cases_context';
-import {
-  generateCaseViewPath,
-  useCaseViewNavigation,
-  useCaseViewParams,
-} from '../../common/navigation';
-import { useCasesTitleBreadcrumbs } from '../use_breadcrumbs';
-
-export interface CaseViewComponentProps {
-  caseId: string;
-  subCaseId?: string;
-  onComponentInitialized?: () => void;
-  actionsNavigation?: CasesNavigation<string, 'configurable'>;
-  ruleDetailsNavigation?: CasesNavigation<string | null | undefined, 'configurable'>;
-  showAlertDetails?: (alertId: string, index: string) => void;
-  useFetchAlertData: (alertIds: string[]) => [boolean, Record<string, Ecs>];
-  /**
-   * A React `Ref` that Exposes data refresh callbacks.
-   * **NOTE**: Do not hold on to the `.current` object, as it could become stale
-   */
-  refreshRef?: MutableRefObject<CaseViewRefreshPropInterface>;
-}
-
-export interface CaseViewProps extends Omit<CaseViewComponentProps, 'caseId' | 'subCaseId'> {
-  timelineIntegration?: CasesTimelineIntegration;
-}
-
-export interface OnUpdateFields {
-  key: keyof Case;
-  value: Case[keyof Case];
-  onSuccess?: () => void;
-  onError?: () => void;
-}
+import { generateCaseViewPath, useCaseViewParams } from '../../common/navigation';
+import { CaseViewPage } from './case_view_page';
+import type { CaseViewProps } from './types';
 
 const MyEuiFlexGroup = styled(EuiFlexGroup)`
   height: 100%;
 `;
 
-export interface CaseComponentProps extends CaseViewComponentProps {
-  fetchCase: UseGetCase['fetchCase'];
-  caseData: Case;
-  updateCase: (newCase: Case) => void;
-}
-
-export const CaseComponent = React.memo<CaseComponentProps>(
-  ({
-    caseData,
-    caseId,
-    fetchCase,
-    onComponentInitialized,
-    actionsNavigation,
-    ruleDetailsNavigation,
-    showAlertDetails,
-    subCaseId,
-    updateCase,
-    useFetchAlertData,
-    refreshRef,
-  }) => {
-    const { userCanCrud } = useCasesContext();
-    const { getCaseViewUrl } = useCaseViewNavigation();
-    useCasesTitleBreadcrumbs(caseData.title);
-
-    const [initLoadingData, setInitLoadingData] = useState(true);
-    const init = useRef(true);
-    const timelineUi = useTimelineContext()?.ui;
-
-    const {
-      caseUserActions,
-      fetchCaseUserActions,
-      caseServices,
-      hasDataToPush,
-      isLoading: isLoadingUserActions,
-      participants,
-    } = useGetCaseUserActions(caseId, caseData.connector.id, subCaseId);
-
-    const { isLoading, updateKey, updateCaseProperty } = useUpdateCase({
-      caseId,
-      subCaseId,
-    });
-
-    // Set `refreshRef` if needed
-    useEffect(() => {
-      let isStale = false;
-
-      if (refreshRef) {
-        refreshRef.current = {
-          refreshCase: async () => {
```

Comment:
Hey! The stale is referring to the state of the component. If the component unmounts (for example a user navigates out of the case single page) then the `isStale` will be true. If in the meantime the `refreshCase` is called then the `fetchCase` will be not called. So, because is based on the component level you cannot have two different states of `isStale` at the same time. What I can see by the merging of the refresh is that a plugin that uses the `refreshRef` has to refresh the whole case again instead of only the user actions and the comments + you have to wait for the case, the metrics, and the user actions to be loaded to refresh again. I am not sure why it was there in the first place tbh.